### DISCUSSION TITLE
Should VistaAlarm_Troube be inverted?

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#48] [Ideas]
Nothing major, mostly aesthetic: I am wondering if it would make more sense to have: binary_sensor.vistaalarm_trouble inverted? off when there is no trouble, on when there is an issue?

**Comment by Dilbert66** [comment]
Huh? Isnt it what it is now?

**Reply by AlexandreUSA** [reply]
Yes, I am may need more sleep. Not sure why it was an issue when I was playing with it before. So weird.

**Reply by AlexandreUSA** [reply]
okay, I am pretty sure it was ON before when there was no trouble. It is now OFF. The checked mark was also "checked". Did something change recently?

**Comment by Dilbert66** [comment]
Just change the device class in the yaml. Home assistant displays status differently depending on the class.
  - platform: template
    id: trouble
    name: "$systemName Trouble"
    device_class: problem

**Comment by Dilbert66** [comment]
I'll double check the conditions that set it. It is always possible that I have a parameter wrong. My test system currently is missing a battery :( so my trouble light is always on. I'll scrounge another battery and run a proper conditions test.

**Reply by AlexandreUSA** [reply]
So weird. It was ON since I started using AlarmVista. However, since I updated to ESPhome 1.9.4 this morning, the sensor flipped to OFF. No faults on my Honeywell keypad. I also thought that was the reason why the device_class: problem was commented out so the check mark was correct.


![image](https://user-images.githubusercontent.com/56739736/123272963-3ad11880-d4d0-11eb-8fea-1153e09188a5.png)

**Reply by AlexandreUSA** [reply]
Something is not right. I just added the device class and reuploaded. Now, I am back at On (Problem).

![image](https://user-images.githubusercontent.com/56739736/123274276-5f79c000-d4d1-11eb-94ea-02162b244983.png)

and another upload, back to Off (Ok)
![image](https://user-images.githubusercontent.com/56739736/123274878-e038bc00-d4d1-11eb-9529-6b3e099e3799.png)

**Reply by AlexandreUSA** [reply]
Please double check the Trouble routines with the battery. As a test, I disconnected the battery to see what would happen but it did not go as planned. The VistaAlarm rebooted then the the AC was showing as unplugged (despite being connected to the outlet).

**Comment by Dilbert66** [comment]
Can you post an excerpt of the log after you disconnect the battery. I'd specifically like to see the F7 cmd which shows the low battery message.?

**Reply by AlexandreUSA** [reply]
I am leaving to Europe tomorrow for a while and my wife does not want me to play with the alarm anymore (she is right on that one). We have someone taking care of the house but I am getting to close to messing it up with the alarm system now. 

When I unplugged the battery, my monitoring company called the house. Glad it was just a fault courtesy call (no emergency dispatch)... I should have put it on test mode first but that made me worry to mess it up before such a long time away from home. When I am back, more than happy to put the system on test and start playing with some of the faults.

**Comment by Dilbert66** [comment]
That version of ESPHOMe is using a buggy esp library or something new. It's causing major crashes on my end.. I recommend you downgrade back to the previous esphome version for now.

Edit: Crash was a compile issue on my end

**Comment by Dilbert66** [comment]
I don't blame you for not wanting to play with the system. I have enough info to dig into the issue. Have a good trip!

**Comment by Dilbert66** [comment]
You need to do a clean build first

**Reply by AlexandreUSA** [reply]
What do you think? I just did the clean build files. All seems fine but should I still go back to ESPhome 1.8.x?

**Comment by Dilbert66** [comment]
It crashes repeatedly on my system even after a clean build. Very odd.  If it's not crashing on yours then leave it for now.  The sensor issue I suspect I know the problem ( a change in my code needed)  but I need to investigate further.

**Comment by Dilbert66** [comment]
I fixed my crash issue by just deleting the temp compile directory and recompiling.  Anyhow i've pushed a new update that should address the incorrect ac/bat/trouble statuses.  It almost seems like new espshome version is not updating the initial field update correctly.  Have to do more tests to know for sure.

**Reply by AlexandreUSA** [reply]
I could not resist and I updated my system with your latest code. Thanks. Everything seems fine. Now, I am done  ;-). I will have 24h to check that everything is stable.

**Comment by Dilbert66** [comment]
So are you saying that now I can take a vacation since you will be away for a while ! :) lol! just kidding.

**Reply by AlexandreUSA** [reply]
For sure. I kept you busy with all my requests. The worse part of all that is that I barely use all the features of the Vista system. However, I like to play with home automation and test random things. That keeps me entertained ;-). Thanks for all the great work. It is awesome to see such a good integration. Next step, all integrated in HA via HACS or directly in HA :-)

**Reply by Dilbert66** [reply]
It was all things that needed to be fixed or improved so tks for your suggestions!

**Comment by AlexandreUSA** [comment]
Closed. Thanks again.
